### PR TITLE
plow: init at 1.3.1

### PR DESCRIPTION
**Pull request by ecklf**
## Description of changes

Adds https://github.com/six-ddc/plow, a high-performance HTTP benchmarking tool.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -5309,6 +5309,11 @@
     githubId = 848000;
     name = "éclairevoyant";
   };
+  ecklf = {
+    github = "ecklf";
+    githubId = 8146736;
+    name = "Florentin Eckl";
+  };
```
Comment (dotlambda) [inline maintainers/maintainer-list.nix:5311]: Please provide another means of communication, e.g. `email`.
See https://github.com/NixOS/rfcs/pull/167.

File: pkgs/by-name/pl/plow/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ lib, buildGoModule, fetchFromGitHub, testers, plow }:
+
+buildGoModule rec {
+  pname = "plow";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "six-ddc";
+    repo = "plow";
+    rev = "c48d821994dbf216e35ea0108232840801952f13";
```
Comment (dotlambda) [inline pkgs/by-name/pl/plow/package.nix:10]:
```suggestion
    rev = "refs/tags/v${version}";
```
The commit doesn't correspond to the tag.

Comment (dotlambda) [reply pkgs/by-name/pl/plow/package.nix]: You have to change `hash` if you change `rev`.

Comment (dotlambda) [reply pkgs/by-name/pl/plow/package.nix]: Set `hash = lib.fakeHash`, try to build and it will tell you the correct hash.

Comment (ecklf) [reply pkgs/by-name/pl/plow/package.nix]: Ok it was the right one 👍

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thanks for your contribution.

Looks like there's a hash mismatch...

```
go: downloading github.com/klauspost/compress v1.15.0
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading golang.org/x/text v0.3.7
Running phase: installPhase
error: hash mismatch in fixed-output derivation '/nix/store/bp0jfjfk235qgbvjz6s2hcxfa5z1gxw2-plow-1.3.1-go-modules.drv':
         specified: sha256-oM0Ko8f4eyC8lRbX16/OiU26nLHcsfhTSNn4x2vyJUo=
            got:    sha256-t2lBPyCn8bu9hLsWmaCGir9egbX0mQR+8kB0RfY7nHE=
error: 1 dependencies of derivation '/nix/store/h9z17zqwwdv9g1dpnajfwlgf8dx8qd2r-plow-1.3.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mmh7nfs4pq7qj5r1q580cv70qjbdk399-plow-1.3.1-test-version.drv' failed to build
error: build of '/nix/store/h9z17zqwwdv9g1dpnajfwlgf8dx8qd2r-plow-1.3.1.drv', '/nix/store/mmh7nfs4pq7qj5r1q580cv70qjbdk399-plow-1.3.1-test-version.drv' failed 
```

**Comment by ecklf** [comment]
> Hi !
> 
> Thanks for your contribution.
> 
> Looks like there's a hash mismatch...
> 
> ```
> go: downloading github.com/klauspost/compress v1.15.0
> go: downloading github.com/valyala/bytebufferpool v1.0.0
> go: downloading golang.org/x/text v0.3.7
> Running phase: installPhase
> error: hash mismatch in fixed-output derivation '/nix/store/bp0jfjfk235qgbvjz6s2hcxfa5z1gxw2-plow-1.3.1-go-modules.drv':
>          specified: sha256-oM0Ko8f4eyC8lRbX16/OiU26nLHcsfhTSNn4x2vyJUo=
>             got:    sha256-t2lBPyCn8bu9hLsWmaCGir9egbX0mQR+8kB0RfY7nHE=
> error: 1 dependencies of derivation '/nix/store/h9z17zqwwdv9g1dpnajfwlgf8dx8qd2r-plow-1.3.1.drv' failed to build
> error: 1 dependencies of derivation '/nix/store/mmh7nfs4pq7qj5r1q580cv70qjbdk399-plow-1.3.1-test-version.drv' failed to build
> error: build of '/nix/store/h9z17zqwwdv9g1dpnajfwlgf8dx8qd2r-plow-1.3.1.drv', '/nix/store/mmh7nfs4pq7qj5r1q580cv70qjbdk399-plow-1.3.1-test-version.drv' failed 
> ```

Fixed, thank you!

**Comment by drupol** [comment]
@ofborg build plow

**Review by drupol — APPROVED** [review]

**Pull request merged**